tests/integration/_conftest/gateway_client_tx_interceptor.py

```diff
@@ -0,0 +1,26 @@
+from typing import Optional
+
+from starknet_py.net.gateway_client import GatewayClient
+from starknet_py.net.client_models import StarknetTransaction
+from starknet_py.net.networks import Network
+
+from .transaction_registry import TransactionRegistry
+
+
+class GatewayClientTxInterceptor(GatewayClient):
+    def __init__(
+        self, net: Network, transaction_registry: Optional[TransactionRegistry] = None
```

Comment:
why make the tx registry optional?

Reply:
[remove optional from transaction registry](https://github.com/software-mansion/protostar/pull/1298/commits/311602d4daa3d52eb9d0d0477fe65a6ddb6930e9)